src/ngx_http_lua_pipe.c

```diff
@@ -688,8 +688,10 @@ ngx_http_lua_ffi_pipe_spawn(ngx_http_lua_ffi_pipe_proc_t *proc,
         /* close listening socket fd */
         ls = ngx_cycle->listening.elts;
         for (i = 0; i < ngx_cycle->listening.nelts; i++) {
-            if (ngx_close_socket(ls[i].fd) == -1) {
-                ngx_log_error(NGX_LOG_EMERG, ngx_cycle->log, ngx_socket_errno,
+            if (ls[i].fd != (ngx_socket_t) -1 &&
+                ngx_close_socket(ls[i].fd) == -1)
+            {
+                ngx_log_error(NGX_LOG_WARN, ngx_cycle->log, ngx_socket_errno,
```

Comment:
I'm glad that we seem to agree on the severity level; if we are to downgrade this log to warn (which I believe is appropriate), don't we ought it too to lower the below logs as well? That is, all calls to `ngx_log_error` that aren't followed by `exit()` I mean.

That said, not a blocker for this fix of course, since the above check in the condition is all this PR should need :+1: